fdbserver/CommitProxyServer.actor.cpp

```diff
@@ -2147,6 +2128,49 @@ ACTOR Future<Void> commitProxyServerCore(CommitProxyInterface proxy,
 						                       /* initialCommit= */ true);
 					}
 
+					// Populate ServerCacheInfo for each shard
+					std::vector<std::pair<MapPair<Key, ServerCacheInfo>, int>> keyInfoData;
+					vector<UID> src, dest;
+					ServerCacheInfo info;
+					for (auto& it : keyServers) {
+						KeyRef k = it.first;
+						if (k != allKeys.end) {
+							decodeKeyServersValue(tag_uid, it.second, src, dest);
+							info.tags.clear();
+							info.src_info.clear();
+							info.dest_info.clear();
+							for (const auto& id : src) {
+								auto storageInfo =
+								    getStorageInfo(id, &commitData.storageCache, commitData.txnStateStore);
+								ASSERT(storageInfo->tag != invalidTag);
+								info.tags.push_back(storageInfo->tag);
+								info.src_info.push_back(storageInfo);
+								if (SERVER_KNOBS->TLOG_NEW_INTERFACE) {
+									// Add storage teams of storage servers
+									ASSERT(storageServerToStorageTeam.count(id));
+									info.storageTeams.insert(storageServerToStorageTeam[id]);
+								}
+							}
+							if (SERVER_KNOBS->TLOG_NEW_INTERFACE) {
+								// A shard can only correspond to single storage team in the primary DC for now
+								ASSERT(info.storageTeams.size() == 1);
+							}
+
+							for (const auto& id : dest) {
+								auto storageInfo =
+								    getStorageInfo(id, &commitData.storageCache, commitData.txnStateStore);
+								ASSERT(storageInfo->tag != invalidTag);
+								info.tags.push_back(storageInfo->tag);
+								info.dest_info.push_back(storageInfo);
+							}
+							uniquify(info.tags);
+							keyInfoData.emplace_back(MapPair<Key, ServerCacheInfo>(k, info), 1);
+						}
+					}
+					// insert keyTag data separately from metadata mutations so that we can do one bulk insert which
+					// avoids a lot of map lookups.
+					commitData.keyInfo.rawInsert(keyInfoData);
+
```

Comment:
I think you can clear `storageServerToStorageTeam` and `keyServers` here and it's necessary to deallocate the memory, since they are state variables.